### PR TITLE
add support for vault secret provider

### PR DESCRIPTION
**Pull request by guysegal**

### CODE REVIEW

**Review by noamokman — CHANGES_REQUESTED** [review]
you have a .DS_Store file commited

File: README.md [hunk 1]
```diff
@@ -21,8 +21,19 @@ A Helm Chart for Dafka Producer
 | metrics.enabled | bool | `true` | should prometheus scrape this server |
 | metrics.path | string | `"/metrics"` | a path prometheus should scrape metrics from |
 | auth.saslUsername | string | `nil` | sasl username |
-| auth.saslPasswordResource | string | `nil` | gcp secret resource for sasl password |
-| auth.useOpaqueSecrets | bool | `true` | mount GCP secrets to Opaque secrets |
-| auth.truststore.truststoreResource | string | `nil` | gcp secret resource for truststore file |
-| auth.truststore.truststorePasswordResource | string | `nil` | gcp secret resource for truststore password |
+| auth.saslPassword | string | `nil` | sasl password (not encrypted) |
+| auth.secrets.useOpaqueSecrets | bool | `true` | should mount secrets to opaque secrets |
+| auth.secrets.useTrustsore | bool | `false` | should use truststore |
```
Comment (noamokman) [inline README.md:26]: should this be under secrets?

File: templates/secrets-provider.yaml [hunk 2]
```diff
@@ -1,30 +1,54 @@
 {{ if not .Values.auth.saslPassword }}
+{{ $provider := "" }}
+{{ if .Values.auth.gcp }}
+{{ $provider = "gcp" }}
+{{ else if .Values.auth.vault }}
+{{ $provider = "vault" }}
+{{ end }}
```
Comment (noamokman) [inline templates/secrets-provider.yaml:7]: so we don't need `auth.secrets.provider`?

File: values.yaml [hunk 3]
```diff
@@ -35,12 +35,34 @@ metrics:
...
+    provider:
+    # -- gcp configurtion
+    gcp:
+      # -- gcp secret resource for sasl password
+      saslPasswordResource:
+      # -- gcp secret resource for truststore file
+      truststoreResource:
+      # -- gcp secret resource for truststore password
+      truststorePasswordResource:
+    # -- vault configurtion
```
Comment (noamokman) [inline values.yaml:55]: this creates a line line for it in the docs, don't think we need it

File: templates/secrets-provider.yaml (hunk 2 above)
Comment (guysegal) [reply templates/secrets-provider.yaml:7]: yeah, I'll make sure it's removed entirely

File: README.md (hunk 1 above)
Comment (guysegal) [reply README.md:26]: currently using truststore is not supported if its not a secret

File: values.yaml (hunk 3 above)
Comment (guysegal) [reply values.yaml]: fixed

**Pull request merged**